src/currencymanager.cpp

```diff
@@ -0,0 +1,274 @@
+    for (auto currency : currencys_) {
+        out += currency.exalt;
+    }
+    return out;
+}
+
+int CurrencyManager::TotalWisdomValue()
+{
+    int out = 0;
+    for (unsigned int i=0; i < wisdoms_.size(); i++) {
+        out += wisdoms_[i] * CurrencyWisdomValue[i];
+    }
+    return out;
+}
+
+void CurrencyManager::ClearCurrency()
+{
+    for(auto &currency : currencys_) {
+        currency.count = 0;
+    }
+    for( auto &wisdom : wisdoms_) {
+        wisdom = 0;
+    }
+}
+
+void CurrencyManager::InitCurrency()
+{
+    std::string value = "";
+    std::string header_csv = "";
+    for (auto curr : CurrencyAsString) {
+        value += "0;";
+        if (curr != "")
+            header_csv += curr + ";";
+    }
+    value.pop_back();//Remove the last ";"
+    header_csv += "Total value; Timestamp";
+    data_.Set("base", value, "currency");
+    data_.Set("last_value",value, "currency");
+    //Create header for .csv
+    QFile file("export_currency.csv");
+    if (file.open(QFile::WriteOnly | QFile::Truncate)){
+        QTextStream out(&file);
+        out  << header_csv.c_str();
+    }
+    else {
+        QLOG_WARN() << "CurrencyManager::ExportCurrency : couldn't open CSV export file ";
+    }
+}
+
+void CurrencyManager::LoadCurrency()
+{
+    //Way easier to use QT function instead of re-implementing a split() in C++
+    QStringList list = QString(data_.Get("base", "", "currency").c_str()).split(";");
+    for (int i=0; i < list.size(); i++) {
+        CurrencyItem curr;
+        curr.count = 0;
+        curr.name = CurrencyAsString[i];
+        //We can't use the toDouble function from QT, because it encodes a double with a ","
+        //Might be related to localisation issue, but anyway it's safer this way
+        curr.base = std::stod(list[i].toStdString());
+        curr.exalt = 0;
+        currencys_.push_back(curr);
+    }
+    for (unsigned int i=0; i < CurrencyWisdomValue.size(); i++)
+    {
+        wisdoms_.push_back(0);
+    }
+
+}
+
+void CurrencyManager::SaveCurrencyBase()
+{
+    std::string value = "";
+    for (auto currency : currencys_) {
+        value += std::to_string(currency.base) + ";";
+    }
+    value.pop_back();//Remove the last ";"
+    data_.Set("base", value, "currency");
+}
+void CurrencyManager::SaveCurrencyValue()
+{
+    std::string value = "";
+    //Useless to save if every count is 0.
+    bool empty = true;
+    for (auto currency : currencys_) {
+        if (currency.name !="")
+            value += std::to_string(currency.count) + ";";
+        if (currency.count !=0)
+            empty = false;
+    }
+    value += std::to_string(TotalExaltedValue());
+    std::string old_value = data_.Get("last_value", "", "currency");
+    if (value != old_value && !empty){
+        std::string timestamp = std::to_string(std::time(nullptr));
+        data_.Set(timestamp, value, "currency");
+        data_.Set("last_value", value, "currency");
+        ExportCurrency(value + ";" + timestamp);
```

Comment:
It can be an idea too, but i saw the export to CSV thing as an automatic feature, who allows you to keep track of your currency without having to do anything. Manual export would kinda defeat this purpose imo